trin-core/src/types/accumulator.rs

```diff
@@ -39,12 +47,24 @@ impl MasterAccumulator {
         Self::default()
     }
 
-    pub fn latest_height(&self) -> u64 {
-        let historical_height = self.epoch_number() * EPOCH_SIZE as u64;
-        historical_height + self.header_record_count()
+    /// Number of the last block to be included in the accumulator
+    pub fn height(&self) -> Option<u64> {
+        let count = self.historical_header_count() + self.current_header_count();
+        match count {
+            0 => None,
+            _ => Some(count - 1),
+        }
+    }
+
+    /// Number of the next block to be included in the accumulator
+    pub fn next_header_height(&self) -> u64 {
+        match self.height() {
+            Some(val) => val + 1,
+            None => 0,
+        }
     }
 
-    fn header_record_count(&self) -> u64 {
+    fn current_header_count(&self) -> u64 {
```

Comment:
```suggestion
    fn current_epoch_len(&self) -> u64 {
```

Reply:
Imo, since we also have the method `historical_header_count()`, I actually prefer `current_header_count()` here. They seem to complement each other nicely

Reply:
I interpret `current_header_count` the same way I interpret `height`, the total number of headers in the accumulator.

If you prefer the `header_count` naming as opposed to `len`, what about `current_epoch_header_count` and `historical_epochs_header_count`?